runtimes/common/src/migrations.rs

```diff
@@ -81,3 +83,209 @@ impl<T: ctype::Config> OnRuntimeUpgrade for AddCTypeBlockNumber<T> {
 		Ok(())
 	}
 }
+
+pub struct MigrateToNewStorageVersion<R>(PhantomData<R>);
+
+impl<R> MigrateToNewStorageVersion<R>
+where
+	R: attestation::Config + pallet_web3_names::Config + public_credentials::Config,
+{
+	fn migrate() -> frame_support::weights::Weight {
+		type AttestationPallet<R> = attestation::Pallet<R>;
+		type Web3NamesPallet<R> = pallet_web3_names::Pallet<R>;
+		type PublicCredentialsPallet<R> = public_credentials::Pallet<R>;
+
+		AttestationPallet::<R>::current_storage_version().put::<AttestationPallet<R>>();
+		// Not an issue with Peregrine, but it is with Spiritnet.
+		Web3NamesPallet::<R>::current_storage_version().put::<Web3NamesPallet<R>>();
+		PublicCredentialsPallet::<R>::current_storage_version().put::<PublicCredentialsPallet<R>>();
+
+		<R as frame_system::Config>::DbWeight::get().writes(4)
+	}
+}
+
+#[cfg(feature = "try-runtime")]
```

Comment:
Did not find a better way to make the dependencies that are only used with `try-runtime` optional.

Reply:
Can you not make the `pre_` and `post_` implementation optional?

Reply:
Yes but you can't import crates only for that. The point of this solution is that the crates for the pallets are imported only when the feature is enabled.

Reply:
Hence the dependency for those can be made optional.

Reply:
ah you are referring to the `where` clause